src/simplify_algebra.cpp

```diff
@@ -851,6 +851,84 @@ struct find_div_const
     }
 };
 
+struct find_zero_div_const
+{
+    auto matcher() const { return match::name("div")(match::arg(1)(match::has_value(0.0f))); }
+
+    [[noreturn]] void apply([[maybe_unused]] module& m,
+                            [[maybe_unused]] const match::matcher_result& r) const
+    {
+        MIGRAPHX_THROW("ERROR: Matched division by zero in pass");
+    }
+};
+
+struct find_unit_ops
+{
+    auto matcher() const
+    {
+        auto mul_1 = match::name("mul")(
+            match::either_arg(0, 1)(match::has_value(1.0f), match::any().bind("x")));
+        auto div_1 =
+            match::name("div")(match::args(match::any().bind("x"), match::has_value(1.0f)));
```

Comment:
When you check `match::has_value` you need to use `skip_broadcasts` otherwise this only will match scalars: `match::skip_broadcasts(match::has_value(1.0f))`.

Reply:
On it. I didn't even realize that. I

Unrelated but general question: Would there be a case where we don't want to cover that case, or as a general rule of them, we should be handling broadcasting always for our matchers?

Reply:
> or as a general rule of them, we should be handling broadcasting always for our matchers?

Well not all matchers. At least, for `has_value` matcher, most times we would want to skip broadcasts, but probably not always.

We could rename the current `has_value` matcher as `has_literal_value` and then make `has_value` do `skip_broadcasts_converts(has_literal_value(...))`. This would probably need some clean up work to remove the redundant `skip_broadcasts`.

Reply:
Yeah it would but seems doable/manageable and would simplify a lot of things. I think we would need to make sure we don't break other things though when we add that no?

For now I'm goin to add the skip_broadcasts statement since it isn't implicit in my has_value calls.

Reply:
@TedThemistokleous Can you add `skip_broadcasts` ?